addons/hr_holidays/models/resource.py

```diff
@@ -32,13 +32,11 @@ def _check_compare_dates(self):
                     raise ValidationError(_('Two public holidays cannot overlap each other for the same working hours.'))
 
     def _get_domain(self, time_domain_dict):
-        domain = []
-        for date in time_domain_dict:
-            domain = expression.OR([domain, [
-                    ('employee_company_id', '=', date['company_id']),
-                    ('date_to', '>', date['date_from']),
-                    ('date_from', '<', date['date_to'])]
-            ])
+        domain = expression.OR([[
+            ('employee_company_id', '=', date['company_id']),
+            ('date_to', '>', date['date_from']),
+            ('date_from', '<', date['date_to']),
+        ] for date in time_domain_dict])
```

Comment:
The structure did not show up to me at first.
```suggestion
        domain = expression.OR([
            [('employee_company_id', '=', date['company_id']),
             ('date_to', '>', date['date_from']),
             ('date_from', '<', date['date_to'])]
            for date in time_domain_dict
        ])
```